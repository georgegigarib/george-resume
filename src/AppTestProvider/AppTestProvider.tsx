import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
