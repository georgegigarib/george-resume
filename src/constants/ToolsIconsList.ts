import JsIcon from '@/assets/icons/tools/Javascript.svg'

export type ToolInfo = {
  name: string
  icon: IconName
  color: string
  bgColor: string
  darkBgColor?: string
  borderColor: string
  darkBorderColor?: string
  darkColor?: string
  svg?: string
  custom?: boolean
}

export type ToolIconProps = {
  tool: IconName
  size?: '5x' | '4x' | '3x' | '2x' | 'xl' | 'lg' | 'md' | 'sm' | 'xs'
}

export enum IconName {
  React = 'react',
  ReactNative = 'reactnative',
  TailwindCSS = 'tailwindcss',
  JavaScript = 'js',
  TypeScript = 'ts-circle',
  VueJs = 'vuejs',
  NextJS = 'nextjs',
  NodeJS = 'nodejs',
  HTML = 'html',
  CSS = 'css',
  Express = 'express',
  Spotify = 'spotify',
  Git = 'git',
  Redis = 'redis',
  Bun = 'bun',
  Docker = 'docker',
  Pinia = 'pinia',
  Sass = 'sass',
  Bootstrap = 'bootstrap',
  Angular = 'angular',
  C = 'c',
  CSharp = 'csharp',
  Django = 'django',
  FramerMotion = 'framer-motion',
  Go = 'go',
  Graphql = 'graphql',
  Javalanguage = 'java',
  Laravel = 'laravel',
  Mui = 'mui',
  Jest = 'jest',
  Vitest = 'vitest',
  Cypress = 'cypress',
  MySQL = 'mysql',
  MongoDb = 'mongodb',
  Php = 'php',
  Postgresql = 'postgresql',
  Python = 'python',
  Redux = 'redux',
  Zustand = 'zustand',
  Svelte = 'svelte',
}

export const toolsIconsList: ToolInfo[] = [
  {
    name: 'React',
    icon: IconName.React,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'React',
    icon: IconName.ReactNative,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Tailwind CSS',
    icon: IconName.TailwindCSS,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#38BDF8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'JavaScript',
    icon: IconName.JavaScript,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
    custom: true,
    svg: JsIcon,
  },
  {
    name: 'Jest',
    icon: IconName.Jest,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MySQL',
    icon: IconName.MySQL,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#38BDF8',
  },
  {
    name: 'Vitest',
    icon: IconName.Vitest,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
  },
  {
    name: 'Cypress',
    icon: IconName.Cypress,
    color: '#000000FF',
    bgColor: '#B4B4B4FF',
    darkBgColor: '#BABABAFF',
    borderColor: '#C7B008FF',
  },
  {
    name: 'TypeScript',
    icon: IconName.TypeScript,
    color: '#3178C6',
    bgColor: '#FAFAFAFF',
    borderColor: '#3178C6',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Next.js',
    icon: IconName.NextJS,
    color: '#0000000',
    bgColor: '#FAFAFAFF',
    borderColor: '#555555FF',
    darkBgColor: '#3E3E3EFF',
    darkColor: '#FFFFFF',
    darkBorderColor: '#BFBFBFFF',
  },
  {
    name: 'Node.js',
    icon: IconName.NodeJS,
    color: '#539E43',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#539E43',
  },
  {
    name: 'HTML',
    icon: IconName.HTML,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'CSS',
    icon: IconName.CSS,
    color: '#264DE4',
    bgColor: '#FAFAFAFF',
    borderColor: '#264DE4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Express',
    icon: IconName.Express,
    color: '#333333',
    bgColor: '#FAFAFAFF',
    borderColor: '#CCCCCC',
  },
  {
    name: 'Spotify',
    icon: IconName.Spotify,
    color: '#1DB954',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#1DB954',
  },
  { name: 'Git', icon: IconName.Git, color: '#F05032', bgColor: '#F5F5F5', borderColor: '#F05032' },
  {
    name: 'Redis',
    icon: IconName.Redis,
    color: '#D82C20',
    bgColor: '#FAFAFAFF',
    borderColor: '#D82C20',
  },
  {
    name: 'Bun',
    icon: IconName.Bun,
    darkColor: '#ffffff',
    color: '#333333',
    bgColor: '#FAFAFAFF',
    darkBorderColor: '#bababa',
    borderColor: '#686767FF',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Docker',
    icon: IconName.Docker,
    color: '#2496ED',
    bgColor: '#FAFAFAFF',
    borderColor: '#2496ED',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Pinia',
    icon: IconName.Pinia,
    color: '#F0C723FF',
    darkBgColor: '#2C2C2C',
    bgColor: '#FAFAFAFF',
    borderColor: '#FFD939',
  },
  {
    name: 'Sass',
    icon: IconName.Sass,
    color: '#CC6699',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#CC6699',
  },
  {
    name: 'Bootstrap',
    icon: IconName.Bootstrap,
    color: '#A877F3FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#7952B3',
  },
  {
    name: 'Angular',
    icon: IconName.Angular,
    color: '#DD0031',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#DD0031',
  },
  {
    name: 'C',
    icon: IconName.C,
    color: '#00599C',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#00599C',
  },
  {
    name: 'C#',
    icon: IconName.CSharp,
    color: '#239120',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#239120',
  },
  {
    name: 'Django',
    icon: IconName.Django,
    color: '#092E20',
    bgColor: '#FAFAFAFF',
    borderColor: '#092E20',
  },
  {
    name: 'Framer Motion',
    icon: IconName.FramerMotion,
    color: '#EC5990',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#EC5990',
  },
  {
    name: 'Go',
    icon: IconName.Go,
    color: '#00ADD8',
    bgColor: '#FAFAFAFF',
    borderColor: '#00ADD8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'GraphQL',
    icon: IconName.Graphql,
    color: '#E10098',
    bgColor: '#FAFAFAFF',
    borderColor: '#E10098',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'java',
    icon: IconName.Javalanguage,
    color: '#007396',
    bgColor: '#FAFAFAFF',
    borderColor: '#007396',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Laravel',
    icon: IconName.Laravel,
    color: '#FF2D20',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF2D20',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Material UI',
    icon: IconName.Mui,
    color: '#0081CB',
    bgColor: '#FAFAFAFF',
    borderColor: '#0081CB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MongoDB',
    icon: IconName.MongoDb,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Vue Js',
    icon: IconName.VueJs,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PHP',
    icon: IconName.Php,
    color: '#959AE1FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#777BB4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PostgreSQL',
    icon: IconName.Postgresql,
    color: '#336791',
    bgColor: '#FAFAFAFF',
    borderColor: '#336791',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Python',
    icon: IconName.Python,
    color: '#3776AB',
    bgColor: '#FAFAFAFF',
    borderColor: '#3776AB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Redux',
    icon: IconName.Redux,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Zustand',
    icon: IconName.Zustand,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Svelte',
    icon: IconName.Svelte,
    color: '#FF3E00',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF3E00',
    darkBgColor: '#2C2C2C',
  },
]
