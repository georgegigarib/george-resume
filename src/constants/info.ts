export const WORK_EMAIL = 'georgegil.work@gmail.com'
export const SOCIAL_LINKS = {
  github: 'https://github.com/georgegigarib',
  linkedin: 'https://www.linkedin.com/in/georgegg/',
  instagram: 'URL_ADDRESS.com/GeorgeGil',
}
