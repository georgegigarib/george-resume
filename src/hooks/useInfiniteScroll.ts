import { useEffect, useRef } from 'react'

const useInfiniteScroll = (containerId: string) => {
  const containerRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const container = containerRef.current
    if (!container) return

    let animationFrame: number
    let scrollPosition = 0

    try {
      // Duplicamos el contenido para permitir un scroll infinito
      const originalContent = container.innerHTML
      container.innerHTML += originalContent

      // Calculamos el ancho total para ajustar el desplazamiento
      const scrollWidth = container.scrollWidth / 2

      const scrollStep = () => {
        scrollPosition += 0.5 // Ajusta la velocidad del desplazamiento aquÃ­
        if (scrollPosition >= scrollWidth) {
          scrollPosition = 0
        }
        container.scrollLeft = scrollPosition
        animationFrame = requestAnimationFrame(scrollStep)
      }

      scrollStep()
    } catch (error) {
      console.error('Error in useInfiniteScroll:', error)
    }

    // Limpieza al desmontar
    return () => {
      if (animationFrame) cancelAnimationFrame(animationFrame)
    }
  }, [containerId])

  return containerRef
}

export default useInfiniteScroll
