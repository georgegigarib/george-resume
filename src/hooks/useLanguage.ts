import { useTranslation } from 'react-i18next';

export const useLanguage = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
    localStorage.setItem('language', lng);
  };

  const loadLanguageFromStorage = () => {
    const savedLanguage = localStorage.getItem('language') || 'en';
    i18n.changeLanguage(savedLanguage);
  };

  return { changeLanguage, loadLanguageFromStorage };
};
