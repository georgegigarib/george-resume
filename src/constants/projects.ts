import image1 from '@/assets/images/projects/movie-app.gif'
import image2 from '@/assets/images/projects/mobile-2.jpeg'
import image3 from '@/assets/images/projects/mobile-1.jpg'
import image4 from '@/assets/images/projects/desktop-app-1.png'
import image5 from '@/assets/images/projects/desktop-app-2.png'
import image6 from '@/assets/images/projects/desktop-app-3.png'
import { IconName } from './ToolsIconsList'

export type ProjectInfoType = {
  src: string
  title: string
  description: string
  url: string
  tools: IconName[]
}

export const mobileProjects: ProjectInfoType[] = [
  {
    src: image1,
    title: 'Movie App',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto y debo alargarlo mas y mas y mas',
    url: 'https://movie-app.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.JavaScript]
  },
  {
    src: image2,
    title: 'Proyecto Mobile 1',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS, IconName.JavaScript]
  },
  {
    src: image3,
    title: 'Proyecto Mobile 2',
    description:
      ' Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto, Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project2.example.com',
    tools: [IconName.React, IconName.TypeScript, IconName.JavaScript]
  },
  {
    src: image1,
    title: 'Proyecto Mobile 2',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto, Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project2.example.com',
    tools: [IconName.React, IconName.TypeScript]
  },
  {
    src: image2,
    title: 'Proyecto Mobile 1',
    description: 'Proyecto con animaciones personalizadas.',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS]
  }
]

export const desktopProjects: ProjectInfoType[] = [
  {
    src: image6,
    title: 'Proyecto Desktop 1',
    description: 'Dashboard avanzado.',
    url: 'https://desktop1.example.com',
    tools: [
      IconName.React,
      IconName.TailwindCSS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git
    ]
  },
  {
    src: image5,
    title: 'Proyecto Desktop 2',
    description:
      'Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git
    ]
  },
  {
    src: image4,
    title: 'Proyecto Desktop 3',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  },
  {
    src: image5,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.NextJS, IconName.NodeJS, IconName.TailwindCSS, IconName.Redis, IconName.Git]
  },
  {
    src: image6,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  },
  {
    src: image6,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  }
]
