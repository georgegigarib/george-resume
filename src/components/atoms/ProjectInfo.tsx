import React from 'react'
import LaunchIcon from '@mui/icons-material/Launch'
import { useIsMobile } from '@/hooks/useIsMobile'
import Marquee from 'react-fast-marquee'
import clsx from 'clsx'
import TechnologyBadge from './TechnologyBadge'
import { IconName } from '@/constants/ToolsIconsList'

type ProjectInfoProps = {
  title: string
  description: string
  url: string
  tools: IconName[]
}

const ProjectInfo: React.FC<ProjectInfoProps> = ({ title, description, url, tools }) => {
  const isMobile = useIsMobile()

  return (
    <div className={clsx(isMobile ? 'flex flex-col mb-8 w-full' : '')}>
      <span className="flex items-center">
        <h2 className="font-bold dark:text-white text-black">{title}</h2>
        <a href={url} target="_blank" rel="noopener noreferrer" className="ml-1">
          <LaunchIcon color="primary" fontSize="small" />
        </a>
      </span>

      <div className="overflow-auto mt-2 mb-2 md:mb-4 max-h-20 md:max-h-[12vh] description-scrollbar">
        <p className="dark:text-white text-black pr-8">{description}</p>
      </div>

      <div className="mt-0">
        {!isMobile && (
          <p className="dark:text-white font-bold text-black mb-2">Used technologies</p>
        )}

        <div className="overflow-auto  md:hidden mt-2 max-h-10 h-14 flex flex-nowrap max-w-[65vw]">
          <Marquee
            gradient={false}
            speed={30}
            direction="left"
            loop={0}
            className={clsx(isMobile ? 'overflow-hidden' : 'flex-wrap flex gap-2')}
          >
            {tools.map((tool, idx) => (
              <TechnologyBadge key={idx} tool={tool} size="md" />
            ))}
          </Marquee>
        </div>

        <div className="overflow-auto md:flex mt-2 max-h-20 md:max-h-[12vh] gap-1 description-scrollbar hidden flex-wrap">
          {tools.map((tool, idx) => (
            <TechnologyBadge key={idx} tool={tool} size="md" />
          ))}
        </div>
      </div>
    </div>
  )
}

export default ProjectInfo
