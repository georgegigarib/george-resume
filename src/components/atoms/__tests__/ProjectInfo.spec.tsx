import { render } from '@testing-library/react'
import { beforeEach, describe, expect, it, vi, Mock } from 'vitest'
import ProjectInfo from '@/components/atoms/ProjectInfo'
import { IconName } from '@/constants/ToolsIconsList'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'
import { useIsMobile } from '@/hooks/useIsMobile'

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}))

class ResizeObserver {
  observe() {}
  unobserve() {}
  disconnect() {}
}

global.ResizeObserver = ResizeObserver

describe('ProjectInfo component', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const renderComponent = (isMobile: boolean, props: any) => {
    ;(useIsMobile as Mock).mockReturnValue(isMobile)
    return render(
      <AppTestProvider>
        <ProjectInfo {...props} />
      </AppTestProvider>
    )
  }

  const defaultDesktopProjectProps = {
    title: 'dev.projects.projects.desktop.exampleProject.title',
    description: 'dev.projects.projects.desktop.exampleProject.description',
    url: 'https://example.com',
    tools: [IconName.React, IconName.TypeScript],
  }

  const defaultMobileProjectProps = {
    ...defaultDesktopProjectProps,
    title: 'dev.projects.projects.mobile.exampleProject.title',
    description: 'dev.projects.projects.mobile.exampleProject.description',
  }

  it('matches the snapshot for desktop view', () => {
    const { asFragment } = renderComponent(false, defaultDesktopProjectProps)
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot for mobile view', () => {
    const { asFragment } = renderComponent(true, defaultMobileProjectProps)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the project title and description for desktop view', () => {
    const { getByText } = renderComponent(false, defaultDesktopProjectProps)
    expect(getByText('Example desktop project')).toBeInTheDocument()
    expect(getByText('Example desktop project description')).toBeInTheDocument()
  })

  it('renders the project title and description for mobile view', () => {
    const { getByText } = renderComponent(true, defaultMobileProjectProps)
    expect(getByText('Example mobile project')).toBeInTheDocument()
    expect(getByText('Example mobile project description')).toBeInTheDocument()
  })

  it('renders the project URL', () => {
    const { getByRole } = renderComponent(false, defaultDesktopProjectProps)
    const link = getByRole('link')
    expect(link).toHaveAttribute('href', 'https://example.com')
  })

  it('renders the technology badges', () => {
    const { getAllByText } = renderComponent(false, defaultDesktopProjectProps)
    const badges = getAllByText('TypeScript')
    expect(badges.length).toBeGreaterThan(0)
  })
})
