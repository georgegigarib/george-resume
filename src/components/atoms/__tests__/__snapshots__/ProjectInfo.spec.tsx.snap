// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectInfo component > matches the snapshot for desktop view 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <span
      class="flex items-center"
    >
      <h2
        class="font-bold dark:text-white text-black"
      >
        Example desktop project
      </h2>
      <a
        class="ml-1"
        href="https://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1m2z4e8-MuiSvgIcon-root"
          data-testid="LaunchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
          />
        </svg>
      </a>
    </span>
    <div
      class="overflow-auto mt-2 mb-2 md:mb-4 max-h-20 md:max-h-[12vh] description-scrollbar"
    >
      <p
        class="dark:text-white text-black pr-8"
      >
        Example desktop project description
      </p>
    </div>
    <div
      class="mt-0"
    >
      <p
        class="dark:text-white font-bold text-black mb-2"
      />
      <div
        class="overflow-auto  md:hidden mt-2 max-h-10 h-14 flex flex-nowrap max-w-[75vw]"
      >
        <div
          class="rfm-marquee-container flex-wrap flex gap-2"
          style="--pause-on-hover: running; --pause-on-click: running; --width: 100%; --transform: none;"
        >
          <div
            class="rfm-marquee"
            style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite; --min-width: 100%;"
          >
            <div
              class="rfm-initial-child-container"
            >
              <div
                class="rfm-child"
                style="--transform: none;"
              >
                <div>
                  <div
                    class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-7lm3m0-MuiChip-root"
                  >
                    <i
                      class="ci ci-react ci-md"
                    />
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                    >
                      React
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="rfm-child"
                style="--transform: none;"
              >
                <div>
                  <div
                    class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1izr7bw-MuiChip-root"
                  >
                    <i
                      class="ci ci-ts-circle ci-md"
                    />
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                    >
                      TypeScript
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rfm-marquee"
            style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite; --min-width: 100%;"
          >
            <div
              class="rfm-child"
              style="--transform: none;"
            >
              <div>
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-7lm3m0-MuiChip-root"
                >
                  <i
                    class="ci ci-react ci-md"
                  />
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                  >
                    React
                  </span>
                </div>
              </div>
            </div>
            <div
              class="rfm-child"
              style="--transform: none;"
            >
              <div>
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1izr7bw-MuiChip-root"
                >
                  <i
                    class="ci ci-ts-circle ci-md"
                  />
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                  >
                    TypeScript
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="overflow-auto md:flex mt-2 max-h-20 md:max-h-[12vh] gap-1 description-scrollbar hidden flex-wrap"
      >
        <div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-7lm3m0-MuiChip-root"
          >
            <i
              class="ci ci-react ci-md"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
            >
              React
            </span>
          </div>
        </div>
        <div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1izr7bw-MuiChip-root"
          >
            <i
              class="ci ci-ts-circle ci-md"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
            >
              TypeScript
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProjectInfo component > matches the snapshot for mobile view 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col mb-8 w-full"
  >
    <span
      class="flex items-center"
    >
      <h2
        class="font-bold dark:text-white text-black"
      >
        Example mobile project
      </h2>
      <a
        class="ml-1"
        href="https://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1m2z4e8-MuiSvgIcon-root"
          data-testid="LaunchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
          />
        </svg>
      </a>
    </span>
    <div
      class="overflow-auto mt-2 mb-2 md:mb-4 max-h-20 md:max-h-[12vh] description-scrollbar"
    >
      <p
        class="dark:text-white text-black pr-8"
      >
        Example mobile project description
      </p>
    </div>
    <div
      class="mt-0"
    >
      <div
        class="overflow-auto  md:hidden mt-2 max-h-10 h-14 flex flex-nowrap max-w-[75vw]"
      >
        <div
          class="rfm-marquee-container overflow-hidden"
          style="--pause-on-hover: running; --pause-on-click: running; --width: 100%; --transform: none;"
        >
          <div
            class="rfm-marquee"
            style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite; --min-width: 100%;"
          >
            <div
              class="rfm-initial-child-container"
            >
              <div
                class="rfm-child"
                style="--transform: none;"
              >
                <div>
                  <div
                    class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-7lm3m0-MuiChip-root"
                  >
                    <i
                      class="ci ci-react ci-md"
                    />
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                    >
                      React
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="rfm-child"
                style="--transform: none;"
              >
                <div>
                  <div
                    class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1izr7bw-MuiChip-root"
                  >
                    <i
                      class="ci ci-ts-circle ci-md"
                    />
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                    >
                      TypeScript
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rfm-marquee"
            style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite; --min-width: 100%;"
          >
            <div
              class="rfm-child"
              style="--transform: none;"
            >
              <div>
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-7lm3m0-MuiChip-root"
                >
                  <i
                    class="ci ci-react ci-md"
                  />
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                  >
                    React
                  </span>
                </div>
              </div>
            </div>
            <div
              class="rfm-child"
              style="--transform: none;"
            >
              <div>
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1izr7bw-MuiChip-root"
                >
                  <i
                    class="ci ci-ts-circle ci-md"
                  />
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                  >
                    TypeScript
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="overflow-auto md:flex mt-2 max-h-20 md:max-h-[12vh] gap-1 description-scrollbar hidden flex-wrap"
      >
        <div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-7lm3m0-MuiChip-root"
          >
            <i
              class="ci ci-react ci-md"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
            >
              React
            </span>
          </div>
        </div>
        <div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1izr7bw-MuiChip-root"
          >
            <i
              class="ci ci-ts-circle ci-md"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
            >
              TypeScript
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
